From d9b04f1894b332c78eb03c26245fb0791dfbbedb Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Thu, 15 Nov 2018 20:46:10 -0700
Subject: [PATCH 09/18] msm8974-common: sepolicy: Label sysfs_socinfo, resolve
 denials

* avc: denied { setattr } for name="soc_iddq" dev="sysfs" ino=5543
  scontext=u:r:init:s0 tcontext=u:object_r:sysfs_socinfo:s0 tclass=file
  permissive=0

Change-Id: Ife248a9cccea19b09b931525606cf4c34344fd9f
---
 sepolicy/common/file_contexts | 1 +
 sepolicy/common/init.te       | 1 +
 2 files changed, 2 insertions(+)

diff --git a/sepolicy/common/file_contexts b/sepolicy/common/file_contexts
index 0dc0946..b2f8a4a 100644
--- a/sepolicy/common/file_contexts
+++ b/sepolicy/common/file_contexts
@@ -48,6 +48,7 @@
 /sys/devices/msm_sdcc\.[0-9]/mmc_host/mmc[0-9]/mmc[0-9]:[0-9]+/block/mmcblk[0-9]/stat        u:object_r:sysfs_disk_stat:s0
 
 # sysfs - cpu/perf
+/sys/devices/system/soc/soc0(/.*)?                      u:object_r:sysfs_socinfo:s0
 /sys/module/msm_pm/modes/cpu[0-9](/.*)?                 u:object_r:sysfs_msm_perf:s0
 
 # sysfs - graphics/panel
diff --git a/sepolicy/common/init.te b/sepolicy/common/init.te
index 0a76cdb..2fafa67 100644
--- a/sepolicy/common/init.te
+++ b/sepolicy/common/init.te
@@ -29,6 +29,7 @@ allow init {
     sysfs_sec_touchkey
     sysfs_sec_tsp
     sysfs_sensors
+    sysfs_socinfo
 }:file setattr;
 
 allow init {
-- 
2.17.1

